scikit-learn/scikit-learn
# [MRG] BUG: add support for non numeric values in MissingIndicator

#### Reference Issues/PRs

closes #13035 

#### What does this implement/fix? Explain your changes.

Add support to non-numeric values in MissingIndicator. I flag this PR as BUG since SimpleImputer is already supporting non-numeric data and that we cannot use the MissingIndicator with the SimpleImputer in 0.20. I think that it should go in 0.20.3

I added 3 tests:

* check that we are raising an error when the numpy string type is raising an error;
* check that we can use string using object dtype;
* check that the missing indicator is working in conjunction with the SimpleImputer.
